### PR TITLE
Added Java Implementation to Investigation [cses-1202]

### PR DESCRIPTION
**Pull request by ChuangSheep**
All test cases passed.

### CODE REVIEW

**Review by nwatx — CHANGES_REQUESTED** [review]
indentation is messed up. Use tabs to indent

**Comment by ChuangSheep** [comment]
opps forgot that. Now fixed.

**Review by nwatx — APPROVED** [review]

File: solutions/cses-1202.mdx [hunk 1]
```diff
@@ -100,4 +100,107 @@ int main() {
...
+		}
+
+		@Override
+		public int compareTo(Pair o) {
+			long dx = f - o.f;
+			if (dx == 0)
+				return s - o.s > 0 ? 1 : -1;
+			else if (dx > 0)
+				return 1;
+			return -1;
```
Comment (nathangong) [inline solutions/cses-1202.mdx:199]: compareTo doesn't have to output only 1 and -1.
```suggestion
			return dx;
```

File: solutions/cses-1202.mdx [hunk 2]
```diff
@@ -100,4 +100,107 @@ int main() {
...
+		public Pair(long x, int y) {
+			f = x;
+			s = y;
+		}
+
+		@Override
+		public int compareTo(Pair o) {
+			long dx = f - o.f;
+			if (dx == 0)
+				return s - o.s > 0 ? 1 : -1;
```
Comment (nathangong) [inline solutions/cses-1202.mdx:196]:
```suggestion
				return s - o.s;
```

File: solutions/cses-1202.mdx [hunk 3]
```diff
@@ -100,4 +100,107 @@ int main() {
...
+			dist[i] = Long.MAX_VALUE;
+			num[i] = 1;
+		}
+
+		// maximum number of flights in the minimum-price route
+		int[] maxf = new int[n];
+		// minimum number of flights in the minimum-price route
+		int[] minf = new int[n];
+
+		Queue<Pair> toVisit = new PriorityQueue<Pair>();
```
Comment (nathangong) [inline solutions/cses-1202.mdx:143]:
```suggestion
		Queue<Pair> toVisit = new PriorityQueue<>();
```

File: solutions/cses-1202.mdx (hunk 1 above)
Comment (ChuangSheep) [reply solutions/cses-1202.mdx:199]: But how do we handle long? We can only return int.

Comment (nathangong) [reply solutions/cses-1202.mdx:199]: whoops

Comment (nathangong) [reply solutions/cses-1202.mdx:199]: you can use `Long.compare` then

Comment (ChuangSheep) [reply solutions/cses-1202.mdx:199]: okay, thanks

**Review by nathangong — APPROVED** [review]

**Review by dongliu0426 — APPROVED** [review]

**Pull request merged**